pkg/api/helpers/download.go

```diff
@@ -0,0 +1,60 @@
+package helpers
+
+import (
+	"github.com/treeverse/lakefs/pkg/api"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+)
+
+// ClientDownload downloads a file using client-side ("direct") access to underlying storage.
+// It requires credentials both to lakeFS and to underlying storage, but considerably reduces
+// the load on the lakeFS server.
+func ClientDownload(ctx context.Context, client api.ClientWithResponsesInterface, repoID, ref, filePath string) (*api.ObjectStats, io.ReadCloser, error) {
```

Comment:
Suggestion: accept a block adapter and avoid a specific s3 implementation.

Reply:
Tempting but currently not really possible.  Adapters are server-side, I probably don't want to bring them into each client.